packages/namekit-react/src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * This TruncatedText component is a duplicate of @namehash/nameguard-react
+ * TruncatedText component. This duplicate will, in the near future, be the
+ * only TruncatedText component in the NameKit libraries. The duplicate is
+ * necessary for the moment so that it is included in main branch while
+ * @namehash/nameguard-react still has its TruncatedText: why is that?
+ *
+ * So any imports of @namehash/nameguard-react TruncatedText component
+ * do not break once this change is merged into main! Once this duplicate
+ * is in main, NameHash Labs will remove the TruncatedText component from
+ * @namehash/nameguard-react and update the imports to use this component
+ * instead, the @namehash/namekit-react TruncatedText.
+ *
+ * A ShortCut story was created to track this migration:
+ * https://app.shortcut.com/ps-web3/story/25261/next-steps-on-displayedname-and-truncatedtext
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import { Tooltip } from "./Tooltip";
+import cc from "classcat";
+
+interface TruncatedTextProps {
+  text: string;
+  maxTooltipWidth?: number;
+  maxDisplayWidth?: number;
+  textStylingClasses?: string;
+  tooltipTextStylingClasses?: string;
+  displayTooltipWhenTextOverflows?: boolean;
+}
+
+const DEFAULT_MAX_DISPLAY_WIDTH = 300;
+const DEFAULT_MAX_TOOLTIP_WIDTH = 400;
```

Comment:
@FrancoAguzzi The idea of a default max tooltip width should move into `Tooltip` and not be here in `TruncatedText`.